test/unit/Account.test.js

```diff
@@ -36,43 +38,61 @@ const mockRes = {
   redirect: (url) => {
     urlToTest = url;
   },
-  json: (resp) => {resp},
+  json: resp => ({ resp }),
 };
 const mockResolve = resp => resp;
 
 describe('`fetchAccountPage`', () => {
-  let fetchAccountPage;
   let requireUser;
   let axiosGet;
+  let mock;
 
   before(() => {
-    fetchAccountPage = sinon.spy(Account, 'fetchAccountPage');
-    requireUser = sinon.stub(User, 'requireUser').callsFake((req, res) => {
-      return ({ redirect: !req.patronTokenResponse.isTokenValid })
-    });
+    requireUser = sinon.stub(User, 'requireUser').callsFake(req => ({ redirect: !req.patronTokenResponse.isTokenValid }));
     axiosGet = sinon.spy(axios, 'get');
+    mock = new MockAdapter(axios);
+    mock
+      .onGet(`${appConfig.legacyCatalog}/dp/patroninfo*eng~Sdefault/6677666/holds`)
+      .reply(200, '<div>some html</div>');
+    mock
+      .onGet(`${appConfig.legacyCatalog}/dp/patroninfo*eng~Sdefault/6677666/items`)
+      .reply(200, '<div>some html</div>');
+    global.store = {
+      getState: () => ({
+        patron: {},
+      }),
+    };
   });
 
   after(() => {
-    fetchAccountPage.restore();
     requireUser.restore();
     axiosGet.restore();
+    global.store = undefined;
+    mock.restore();
   });
 
   beforeEach(() => {
     axiosGet.reset();
-  })
+  });
 
   describe('patron not logged in', () => {
     it('should not make axios request', () => {
-      fetchAccountPage(renderMockReq('items', { isTokenValid: false }), mockRes, mockResolve);
+      Account.fetchAccountPage(renderMockReq('items', { isTokenValid: false }), mockRes, mockResolve);
       expect(axiosGet.notCalled).to.equal(true);
     });
   });
 
   describe('does not receive valid value from "req.params.content"', () => {
+    before(() => {
+      global.store = {
+        getState: () => ({
+          patron,
+        }),
+      };
+    });
+
     it('should redirect', () => {
-      fetchAccountPage(renderMockReq('blahblah'), mockRes, mockResolve);
+      Account.fetchAccountPage(renderMockReq('blahblah'), mockRes, mockResolve);
 
       expect(urlToTest).to.equal('/research/collections/shared-collection-catalog/account');
```

Comment:
This appears to have been an existing mechanism, but may I suggest a small change to reduce the chance of collision with other tests?

```
    it('should redirect', () => {
      let redirectedTo = null;

      Account.fetchAccountPage(renderMockReq('blahblah'), { json: () => null, redirect: (url) => redirectedTo = url }, mockResolve);

      expect(redirectedTo).to.equal('/research/collections/shared-collection-catalog/account');
    });
```

Currently you're using a common mock resolver, which writes to root-level `urlToTest` var when it encounters a redirect. It looks like you currently have only one test making expectations on redirects, so there's no risk currently. But if you had one other test that potentially generated a redirect, it seems possible that hitting a redirect in the first test may cause `urlToTest` to be set, which might be mistaken as the result of the latter test. In general we should ensure the context around the tests is reset between tests so that one test can not change the outcome of another.